### PR TITLE
Add a regression test for #13160

### PR DESCRIPTION
**Pull request by parjong**

### CODE REVIEW

**Comment by parjong** [comment]
```
D:\j\workspace\x64_release_w---0575cb46\bin\tests\Windows_NT.x64.Release\TestWrappers\JIT.jit64\JIT.jit64.XUnitWrapper.cs(28597): error : JIT_jit64._opt_cse_VolatileTest_op_add_VolatileTest_op_add_._opt_cse_VolatileTest_op_add_VolatileTest_op_add_cmd [FAIL] [D:\j\workspace\x64_release_w---0575cb46\tests\runtest.proj]
08:35:34         
08:35:34   
08:35:34   Return code:      1
08:35:34   Raw output file:      D:\j\workspace\x64_release_w---0575cb46\bin\tests\Windows_NT.x64.Release\Reports\JIT.jit64\opt\cse\VolatileTest_op_add\VolatileTest_op_add.output.txt
08:35:34   Raw output:
08:35:34   BEGIN EXECUTION
08:35:34          "D:\j\workspace\x64_release_w---0575cb46\tests\..\bin\tests\Windows_NT.x64.Release\Tests\Core_Root\corerun.exe" VolatileTest_op_add.exe 
08:35:34         this test is designed to hang if jit cse doesnt honor volatile
08:35:34         Test 1 passed after 8332906 tries
08:35:34         Test 2 passed after 138 tries
08:35:34         Test 3 passed after 235 tries
08:35:34         Expected: 100
08:35:34         Actual: -1073740791
08:35:34         END EXECUTION - FAILED
08:35:34         FAILED
08:35:34         Test Harness Exitcode is : 1
```

**Comment by parjong** [comment]
@dotnet-bot test Windows_NT x64 Release Priority 1 Build and Test please

**Comment by parjong** [comment]
@dotnet-bot test Tizen armel Cross Debug Build please

File: tests/src/CoreMangLib/system/delegate/regressions/Github_13160/Github_13160.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    static int Main()
+    {
+        Program p = new Program();
+
+        Action d1 = p.VirtualMethod;
+        Action d2 = p.VirtualMethod;
+
+        if (d1.Equals(d2))
```
Comment (jkotas) [inline tests/src/CoreMangLib/system/delegate/regressions/Github_13160/Github_13160.cs]: `d1` should be equal `d2` as well here. Why not test it?

Comment (parjong) [inline tests/src/CoreMangLib/system/delegate/regressions/Github_13160/Github_13160.cs]: I though that that equality may not hold. I'll updated PR.

**Comment by parjong** [comment]
@dotnet-bot test Tizen armel Cross Debug Build please

**Comment by jkotas** [comment]
Thanks

**Pull request merged**